### PR TITLE
Make TxGraph::calculate_fee return 0 for coinbase

### PR DESCRIPTION
**Pull request by LLFourn**
And fix a panic. Arguable this should return a negative value but I've left the API the same for now.

### CODE REVIEW

File: bdk_chain/tests/test_tx_graph.rs [hunk 1]
```diff
@@ -141,3 +141,97 @@ fn insert_tx_can_retrieve_full_tx_from_graph() {
...
+    assert_eq!(graph.calculate_fee(&tx), Some(100));
+
+    tx.input.remove(2);
+
+    // fee would be negative
+    assert_eq!(graph.calculate_fee(&tx), Some(-200));
+}
+
+#[test]
+fn test_calculate_fee_negative_fee() {}
```
Comment (rajarshimaitra) [inline bdk_chain/tests/test_tx_graph.rs:220]: just to note that this test is already covered in the previous one.

File: bdk_chain/tests/test_tx_graph.rs [hunk 2]
```diff
@@ -141,3 +141,97 @@ fn insert_tx_can_retrieve_full_tx_from_graph() {
...
+
+#[test]
+fn test_calculate_fee_on_coinbase() {
```
Comment (rajarshimaitra) [inline bdk_chain/tests/test_tx_graph.rs:223]: just for the sake of completeness maybe we can also add for the `None` case, where we are missing one/many inputs.

**Review by rajarshimaitra — COMMENTED** [review]
tACK 60250862631015c2efa9b93a911ad4c1744f4da7

File: bdk_chain/tests/test_tx_graph.rs [hunk 3]
```diff
@@ -141,3 +141,97 @@ fn insert_tx_can_retrieve_full_tx_from_graph() {
     graph.insert_tx(tx.clone());
     assert_eq!(graph.tx(tx.txid()), Some(&tx));
 }
+
+#[test]
+fn test_calcualte_fee() {
```
Comment (evanlinjin) [inline bdk_chain/tests/test_tx_graph.rs:146]:
```suggestion
fn test_calculate_fee() {
```

**Review by evanlinjin — APPROVED** [review]
ACK 60250862631015c2efa9b93a911ad4c1744f4da7

Just a NIT and Raj's suggestions on tests

**Review by evanlinjin — APPROVED** [review]
ACK 181e0cc4810e56d21d8b7e97f23965fd8096a356

**Pull request merged**